---
title: "Junior Paper"
author: "Andres Blanco Bonilla"
date: "2023-04-04"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Code for Junior Paper

```{r}
library("TCGAbiolinks")
library("limma")
library("edgeR")
library("glmnet")
library("factoextra")
library("FactoMineR")
library("caret")
library("SummarizedExperiment")
library("gplots")
library("survival")
library("survminer")
library("RColorBrewer")
library("gProfileR")
library("genefilter")
```
Failed Query (not enough RAM)
```{r}
#query_TCGA = GDCquery(
#  project = "TCGA-BRCA",
#  data.category = "Transcriptome Profiling",
#  experimental.strategy = "RNA-Seq",
#  data.type = "Gene Expression Quantification",
#  workflow.type = "STAR - Counts",
#  sample.type = c("Solid Tissue Normal"))

#lihc_res = getResults(query_TCGA)
#head(lihc_res)
#summary(factor(lihc_res$sample_type))
#GDCdownload(query = query_TCGA)
#tcga_data = GDCprepare(query_TCGA)
#BRCAMatrix <- assay(tcga_data)
#BRCAMatrix
#rowData(tcga_data)
```
Constructing BRCA data (with ancestry) csv file manually
```{r}
library(RTCGA.rnaseq)
library(dplyr)

brca_expression_data <- BRCA.rnaseq
#dim(brca_expression_data)
#head(brca_expression_data)
#normal_samples <- filter(brca_expression_data, #substr(brca_expression_data$bcr_patient_barcode, 14, 15) == "11")
#dim(normal_samples)
#head(normal_samples)

brca_genetic_ancestry <- read.csv("/Users/andres/Downloads/jp/ancestry.csv", na.strings = "NNN")
dim(brca_genetic_ancestry)
head(brca_genetic_ancestry)
normal_aa_samples <- filter(brca_genetic_ancestry, brca_genetic_ancestry$genetic_ancestry == "AA")
normal_ea_samples <- filter(brca_genetic_ancestry, brca_genetic_ancestry$genetic_ancestry == "EA")
dim(normal_aa_samples)
dim(normal_ea_samples)

ancestries <- c(nrow(brca_expression_data))

for (i in 1:nrow(brca_expression_data)) {
  for (j in 1:nrow(brca_genetic_ancestry)) {
    if (grepl(brca_genetic_ancestry$bcr_patient_barcode[j],brca_expression_data$bcr_patient_barcode[i], fixed = TRUE)){
      ancestries[i] = brca_genetic_ancestry$genetic_ancestry[j]
      break
      }
  }
}
brca_full_data <- brca_expression_data
brca_full_data['genetic_ancestry'] <- ancestries
brca_full_data <- brca_full_data[,c(1,20533,2,3:(ncol(brca_full_data) - 1))]
head(brca_full_data)
dim(brca_full_data)

write.csv(brca_full_data, "brca_data.csv", row.names=FALSE)

```
Add survival times and vital status to csv for survival analysis
```{r}
library(RTCGA.clinical)
survival_data <- survivalTCGA(BRCA.clinical)
head(survival_data)

times_vector <- c(nrow(brca_full_data))
vital_status_vector <- c(nrow(brca_full_data))
for (i in 1:nrow(brca_full_data)) {
  for (j in 1:nrow(survival_data)) {
    if (grepl(survival_data$bcr_patient_barcode[j],brca_full_data$bcr_patient_barcode[i], fixed = TRUE)){
      times_vector[i] = survival_data$times[j]
      vital_status_vector[i] = survival_data$patient.vital_status[j]
      }
  }
}
brca_full_data['times'] <- times_vector
brca_full_data <- brca_full_data[,c(1,2,20534, 3:(ncol(brca_full_data) - 1))]
brca_full_data['patient.vital_status'] <- vital_status_vector
brca_full_data <- brca_full_data[,c(1,2,20535, 3:(ncol(brca_full_data) - 1))]
head(brca_full_data)

write.csv(brca_full_data, "brca_full_data.csv", row.names=FALSE)
```
Add tissue type label
```{r}
library(tidyverse)
brca_full_data <- read.csv("brca_full_data.csv", na.strings = "NNN")
tissue_type <- substr(brca_expression_data$bcr_patient_barcode, 14, 15) == "01"
tissue_type <- replace(tissue_type,tissue_type==TRUE,"Primary Solid Tumor")
tissue_type <- replace(tissue_type,tissue_type==FALSE,"Solid Tissue Normal")
head(tissue_type)
tissue_type

brca_full_data['tissue_type'] <- tissue_type

write.csv(brca_full_data, "brca_full_data_with_type.csv", row.names=FALSE)

normal_samples <- filter(brca_full_data, substr(brca_full_data$bcr_patient_barcode, 14, 15) == "11")

normal_aa_samples <- filter(normal_samples, normal_samples$genetic_ancestry == "AA")
normal_ea_samples <- filter(normal_samples, normal_samples$genetic_ancestry == "EA")


dim(normal_aa_samples)
dim(normal_ea_samples)
```
All
```{r}
library(caret)
library(caTools)
library(dplyr)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode, times, patient.vital_status))
brca_exp_data <- subset(brca_exp_data, genetic_ancestry %in% c("EA", "AA"))
brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)
set.seed(757)
split <- sample.split(brca_exp_data$genetic_ancestry, SplitRatio = 0.80)
brca_exp_data <- subset(brca_exp_data, select = -c(genetic_ancestry))
all_training_set <- subset(brca_exp_data, split == TRUE)
all_test_set <- subset(brca_exp_data, split == FALSE)
fit_svm_all_brca <- train(tissue_type ~ ., data = all_training_set, method = "svmLinear")
predict_svm_all_brca <- predict(fit_svm_all_brca, all_test_set)
confusionMatrix(reference=all_test_set$tissue_type, data = predict_svm_all_brca, mode = "everything")
predictions_all <- as.character(predict_svm_all_brca)

rownames(all_test_set)[which(all_test_set$tissue_type != predictions_all)]
all_aa_test <- subset(brca_full_data, bcr_patient_barcode %in% rownames(all_test_set))
length(which(all_aa_test$genetic_ancestry == "AA"))

```
EA/AA
```{r}
#brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
rownames(brca_full_data) <- patient_ids
normal_ea_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "EA"))
normal_aa_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "AA"))

normal_aa_samples <- subset(normal_aa_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
normal_ea_samples <- subset(normal_ea_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
normal_aa_samples$tissue_type <- as.factor(normal_aa_samples$tissue_type)
normal_ea_samples$tissue_type <- as.factor(normal_ea_samples$tissue_type)

fit_svm_ea_brca <- train(tissue_type ~ ., data = normal_ea_samples, method = "svmLinear")
predict_svm_aa_brca <- predict(fit_svm_ea_brca, normal_aa_samples)
confusionMatrix(reference=normal_aa_samples$tissue_type, data = predict_svm_aa_brca, mode = "everything")

predictions_ea_aa <- as.character(predict_svm_aa_brca)
rownames(normal_aa_samples)[which(normal_aa_samples$tissue_type != predictions_ea_aa)]

```
EA Only
```{r}
library(caret)
library(caTools)
library(dplyr)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")

ea_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "EA"))
brca_ea_exp_data <- subset(ea_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
brca_ea_exp_data$tissue_type <- as.factor(brca_ea_exp_data$tissue_type)

split <- sample.split(brca_ea_exp_data$tissue_type, SplitRatio = 0.80)
ea_training_set <- subset(brca_ea_exp_data, split == TRUE)
ea_test_set <- subset(brca_ea_exp_data, split == FALSE)
fit_svm_ea_brca <- train(tissue_type ~ ., data = ea_training_set, method = "svmLinear")
predict_svm_ea_brca <- predict(fit_svm_ea_brca, ea_test_set)
confusionMatrix(reference=ea_test_set$tissue_type, data = predict_svm_ea_brca, mode = "everything")

predictions_ea <- as.character(predict_svm_ea_brca)
rownames(ea_test_set)[which(ea_test_set$tissue_type != predictions_ea)]

```
AA Only
```{r}
library(caret)
library(caTools)
library(dplyr)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")

aa_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "AA"))
brca_aa_exp_data <- subset(aa_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
brca_aa_exp_data$tissue_type <- as.factor(brca_aa_exp_data$tissue_type)

split <- sample.split(brca_aa_exp_data$tissue_type, SplitRatio = 0.70)
aa_training_set <- subset(brca_aa_exp_data, split == TRUE)
aa_test_set <- subset(brca_aa_exp_data, split == FALSE)
fit_svm_aa_brca <- train(tissue_type ~ ., data = aa_training_set, method = "svmLinear")
predict_svm_aa_brca <- predict(fit_svm_aa_brca, aa_test_set)
confusionMatrix(reference=aa_test_set$tissue_type, data = predict_svm_aa_brca, mode = "everything")

predictions_aa <- as.character(predict_svm_aa_brca)
rownames(aa_test_set)[which(aa_test_set$tissue_type != predictions_aa)]


```
EA+AA?
```{r}
aae_training_set <- rbind(ea_training_set, aa_training_set, aa_test_set)
fit_svm_aae_brca <- train(tissue_type ~ ., data = aae_training_set, method = "svmLinear")
predict_svm_aae_brca <- predict(fit_svm_aae_brca, ea_test_set)
confusionMatrix(reference=ea_test_set$tissue_type, data = predict_svm_aae_brca, mode = "everything")

predictions_aae <- as.character(predict_svm_aae_brca)
rownames(ea_test_set)[which(ea_test_set$tissue_type != predictions_aae)]

```
Glmnet
All
```{r}
library(caret)
library(caTools)
library(dplyr)
library(glmnet)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode, times, patient.vital_status))
brca_exp_data <- subset(brca_exp_data, genetic_ancestry %in% c("EA", "AA"))
brca_exp_data <- subset(brca_exp_data, select = -c(genetic_ancestry))
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)

set.seed(13)
split <- sample.split(brca_exp_data$tissue_type, SplitRatio = 0.80)
all_training_set <- subset(brca_exp_data, split == TRUE)
all_test_set <- subset(brca_exp_data, split == FALSE)
y_train <- as.matrix(all_training_set$tissue_type)
x_train <- as.matrix(subset(all_training_set, select = -c(tissue_type)))
res = cv.glmnet(
  x = x_train,
  y = y_train,
  alpha = 0.5,
  family = "binomial"
)
y_test <- as.matrix(all_test_set$tissue_type)
x_test <- as.matrix(subset(all_test_set, select = -c(tissue_type)))

y_pred = predict(res, newx=x_test, type="class", s="lambda.min")
confusion_matrix = table(y_pred, y_test)
print(confusion_matrix)
rownames(y_pred)[which(y_test != y_pred)]

all_aa_test_g <- subset(brca_full_data, bcr_patient_barcode %in% rownames(all_test_set))
length(which(all_aa_test_g$genetic_ancestry == "AA"))
```
EA/AA
```{r}
library(caret)
library(caTools)
library(dplyr)
library("glmnet")
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
normal_ea_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "EA"))

normal_aa_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "AA"))

normal_aa_samples <- subset(normal_aa_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
normal_ea_samples <- subset(normal_ea_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)

all_training_set <- normal_ea_samples
all_test_set <- normal_aa_samples
y_train <- as.matrix(all_training_set$tissue_type)
x_train <- as.matrix(subset(all_training_set, select = -c(tissue_type)))
class(x_train)
class(y_train)
res = cv.glmnet(
  x = x_train,
  y = y_train,
  alpha = 0.5,
  family = "binomial"
)
y_test2 <- as.matrix(all_test_set$tissue_type)
x_test2 <- as.matrix(subset(all_test_set, select = -c(tissue_type)))

y_pred2 = predict(res, newx=x_test2, type="class", s="lambda.min")
confusion_matrix2 = table(y_pred2, y_test2)
print(confusion_matrix2)
rownames(y_pred2)[which(y_test2 != y_pred2)]

```
EA Only
```{r}
library(caret)
library(caTools)
library(dplyr)
library("glmnet")
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")

normal_ea_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "EA"))
normal_ea_samples <- subset(normal_ea_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
split <- sample.split(normal_ea_samples$tissue_type, SplitRatio = 0.80)
all_training_set_ea <- subset(normal_ea_samples, split == TRUE)
all_test_set_ea <- subset(normal_ea_samples, split == FALSE)
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)

y_train_ea <- as.matrix(all_training_set_ea$tissue_type)
x_train_ea <- as.matrix(subset(all_training_set_ea, select = -c(tissue_type)))

res = cv.glmnet(
  x = x_train_ea,
  y = y_train_ea,
  alpha = 0.5,
  family = "binomial"
)
y_test_ea <- as.matrix(all_test_set_ea$tissue_type)
x_test_ea <- as.matrix(subset(all_test_set_ea, select = -c(tissue_type)))

y_pred_ea = predict(res, newx=x_test_ea, type="class", s="lambda.min")
confusion_matrix_ea = table(y_pred_ea, y_test_ea)
print(confusion_matrix_ea)
rownames(y_pred_ea)[which(y_test_ea != y_pred_ea)]

```
AA Only
```{r}
library(caret)
library(caTools)
library(dplyr)
library("glmnet")
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")

normal_aa_samples <- filter(brca_full_data, (brca_full_data$genetic_ancestry == "AA"))

normal_aa_samples <- subset(normal_aa_samples, select = -c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
split <- sample.split(normal_aa_samples$tissue_type, SplitRatio = 0.80)
aa_training_set <- subset(normal_aa_samples, split == TRUE)
aa_test_set <- subset(normal_aa_samples, split == FALSE)
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)


y_train_aa <- as.matrix(aa_training_set$tissue_type)
x_train_aa <- as.matrix(subset(aa_training_set, select = -c(tissue_type)))

res = cv.glmnet(
  x = x_train_aa,
  y = y_train_aa,
  alpha = 0.5,
  family = "binomial"
)
y_test_aa <- as.matrix(aa_test_set$tissue_type)
x_test_aa <- as.matrix(subset(aa_test_set, select = -c(tissue_type)))

y_pred_aa = predict(res, newx=x_test_aa, type="class", s="lambda.min")
confusion_matrix_aa = table(y_pred_aa, y_test_aa)
print(confusion_matrix_aa)
```
EA + AA
```{r}
library(caret)
library(caTools)
library(dplyr)
library("glmnet")
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")

aae_x_train <- rbind(x_train_ea, x_train_aa, x_test_aa)
aae_y_train <- rbind(y_train_ea, y_train_aa, y_test_aa)
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)

res = cv.glmnet(
  x = aae_x_train,
  y = aae_y_train,
  alpha = 0.5,
  family = "binomial"
)
y_test_aae <- y_test_ea
x_test_aae <- x_test_ea

y_pred_aae = predict(res, newx=x_test_aae, type="class", s="lambda.min")
confusion_matrix_aae = table(y_pred_aae, y_test_aae)
print(confusion_matrix_aae)
rownames(y_pred_aae)[which(y_test_aae != y_pred_aae)]
```
EA + AA/AA
```{r}
library(caret)
library(caTools)
library(dplyr)
library("glmnet")
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")

eea_x_train <- rbind(x_train_ea, x_train_aa, x_test_ea)
eea_y_train <- rbind(y_train_ea, y_train_aa, y_test_ea)
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)

res = cv.glmnet(
  x = eea_x_train,
  y = eea_y_train,
  alpha = 0.5,
  family = "binomial"
)
y_test_eea <- y_test_aa
x_test_eea <- x_test_aa

y_pred_eea = predict(res, newx=x_test_eea, type="class", s="lambda.min")
confusion_matrix_eea = table(y_pred_eea, y_test_eea)
print(confusion_matrix_eea)
rownames(y_pred_eea)[which(y_test_eea != y_pred_eea)]
```
Blockforest
```{r}
library(caTools)
library(dplyr)
library(blockForest)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode, genetic_ancestry))
rownames(brca_exp_data) <- patient_ids
split <- sample.split(brca_exp_data$tissue_type, SplitRatio = 0.80)
all_training_set <- subset(brca_exp_data, split == TRUE)
all_test_set <- subset(brca_exp_data, split == FALSE)
y_train_s <- cbind(all_training_set$times, all_training_set$patient.vital_status)
x_train_s <- subset(all_training_set, select = -c(tissue_type))
x_test_s <- subset(all_test_set, select = -c(tissue_type))
block_list <- list(c(1:ncol(x_train_s)))

blockforobj <- blockfor(x_train_s, y_train_s, blocks = block_list)
predres <- predict(blockforobj$forest, data = x_test_s)
rowSums(predres$chf)
```
Next
```{r}
library("survcomp")
library(blockForest)
blockforobj <- readRDS("block_forest_all.RDS")
blockforpred <- readRDS("block_forest_pred_all.RDS")
pred_results <- read.csv("block_forest_pred_all.csv")
pred_times <- pred_results$V2
actual_times <- brca_full_data[brca_full_data$bcr_patient_barcode %in% pred_results$V1, "times"]
patient_status <- brca_full_data[brca_full_data$bcr_patient_barcode %in% pred_results$V1, "patient.vital_status"]
c_index <- concordance.index(x=pred_times, surv.time = actual_times, surv.event = patient_status)
c_index$c.index

```
N
```{r}
library(caTools)
library(dplyr)
library(blockForest)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode))
rownames(brca_exp_data) <- patient_ids
normal_ea_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry == "EA"))
normal_aa_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry == "AA"))
all_training_set_ea <- subset(normal_ea_samples, select = -c(genetic_ancestry))
all_test_set_aa <- subset(normal_aa_samples, select = -c(genetic_ancestry))
y_train_s_ea <- cbind(all_training_set_ea$times, all_training_set_ea$patient.vital_status)
x_train_s_ea <- subset(all_training_set_ea, select = -c(tissue_type))
x_test_s_aa <- subset(all_test_set_aa, select = -c(tissue_type))
block_list <- list(c(1:ncol(x_train_s_ea)))

blockforobj_ea_aa <- blockfor(x_train_s_ea, y_train_s_ea, blocks = block_list)
predres_ea_aa <- predict(blockforobj$forest, data = x_test_s_aa)
pred_results_ea_aa <- cbind(row.names(x_test_s_aa), rowSums(predres$chf))
pred_times <- pred_results_ea_aa$V2
actual_times <- brca_full_data[brca_full_data$bcr_patient_barcode %in% pred_results_ea_aa$V1, "times"]
patient_status <- brca_full_data[brca_full_data$bcr_patient_barcode %in% pred_results_ea_aa$V1, "patient.vital_status"]
c_index_ea_aa <- concordance.index(x=pred_times, surv.time = actual_times, surv.event = patient_status)
c_index_ea_aa$c.index
```
All survival
```{r}
library(caTools)
library(dplyr)
library(blockForest)
library(survival)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode, tissue_type))
rownames(brca_exp_data) <- patient_ids
brca_exp_data <- subset(brca_exp_data, genetic_ancestry %in% c("EA", "AA"))
brca_exp_data <- subset(brca_exp_data, select = -c(genetic_ancestry))

set.seed(757)
split <- sample.split(brca_exp_data$patient.vital_status, SplitRatio = 0.80)
x_train_rf <- subset(brca_exp_data, split == TRUE)
x_test_rf <- subset(brca_exp_data, split == FALSE)
x_train_rf <- subset(x_train_rf, times >= 0)
x_test_rf <- subset(x_test_rf, times >= 0)
rf_all <- rfsrc(formula =
Surv(times,patient.vital_status)~., data = x_train_rf)
rf_pred_all <- predict(rf_all, newdata = x_test_rf)
rf_pred_all
significant_genes <- c("times", "patient.vital_status", "CD24.100133941", "PRRG1.5638", "IQSEC3.440073","MRGPRX1.259249","RCC2.55920","CASP8.841")
x_train_cox <- subset(x_train_rf, select = significant_genes)
x_test_cox <- subset(x_test_rf, select = significant_genes)
cox_all <- coxph(formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_train_cox, x = TRUE, y = TRUE)
cox_all_pec <- pec(cox_all, formula = Surv(times,patient.vital_status)~ CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_test_cox)
crps(cox_all_pec)
cindex(cox_all, formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_test_cox, cens.model = "cox")
```
Next
```{r}
library(caTools)
library(dplyr)
library(survival)
library(randomForestSRC)
library(pec)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode))
rownames(brca_exp_data) <- patient_ids
normal_ea_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry == "EA"))
normal_aa_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry == "AA"))
normal_aa_samples <- subset(normal_aa_samples, select = -c(genetic_ancestry))
normal_ea_samples <- subset(normal_ea_samples, select = -c(genetic_ancestry))
x_train_rf_ea <- subset(normal_ea_samples, select = -c(tissue_type))
x_test_rf_aa <- subset(normal_aa_samples, select = -c(tissue_type))
rf_ea_aa <- rfsrc(formula =
Surv(times,patient.vital_status)~., data = x_train_rf_ea)
rf_pred_ea_aa <- predict(rf_ea_aa, newdata = x_test_rf_aa)
rf_pred_ea_aa
significant_genes <- c("times", "patient.vital_status", "CD24.100133941", "PRRG1.5638", "IQSEC3.440073","MRGPRX1.259249","RCC2.55920","CASP8.841")
x_train_cox_ea <- subset(x_train_rf_ea, select = significant_genes)
x_test_cox_aa <- subset(x_test_rf_aa, select = significant_genes)
cox_ea_aa <- coxph(formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_train_cox_ea, x = TRUE, y = TRUE)
cox_ea_aa_pec <- pec(cox_ea_aa, formula = Surv(times,patient.vital_status)~ CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_test_cox_aa)
crps(cox_ea_aa_pec)
cindex(cox_ea_aa, formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_test_cox_aa, cens.modeL
= "cox")
```
EA only
```{r}
library(caTools)
library(dplyr)
library(blockForest)
library(survival)
library(randomForestSRC)
library(pec)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode))
rownames(brca_exp_data) <- patient_ids
normal_ea_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry == "EA"))
normal_ea_samples <- subset(normal_ea_samples, select = -c(genetic_ancestry))
set.seed(757)
split <- sample.split(normal_ea_samples$patient.vital_status, SplitRatio = 0.80)
x_train_rf_ea <- subset(normal_ea_samples, split == TRUE)
x_test_rf_ea <- subset(normal_ea_samples, split == FALSE)
x_train_rf_ea <- subset(x_train_rf_ea, select = -c(tissue_type))
x_test_rf_ea <- subset(x_test_rf_ea, select = -c(tissue_type))
x_train_rf_ea <- subset(x_train_rf_ea, times >= 0)
x_test_rf_ea <- subset(x_test_rf_ea, times >= 0)
rf_ea <- rfsrc(formula =
Surv(times,patient.vital_status)~., data = x_train_rf_ea)
rf_pred_ea <- predict(rf_ea, newdata = x_test_rf_ea)
rf_pred_ea
significant_genes <- c("times", "patient.vital_status", "CD24.100133941", "PRRG1.5638", "IQSEC3.440073","MRGPRX1.259249","RCC2.55920","CASP8.841")
x_train_cox_ea <- subset(x_train_rf_ea, select = significant_genes)
x_test_cox_ea <- subset(x_test_rf_ea, select = significant_genes)
cox_ea <- coxph(formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_train_cox_ea, x = TRUE, y = TRUE)
cox_ea_pec <- pec(cox_ea, formula = Surv(times,patient.vital_status)~ CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_test_cox_ea)
crps(cox_ea_pec)
cindex(cox_ea, formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
MRGPRX1.259249 + RCC2.55920 + CASP8.841, data = x_test_cox_ea, cens.model = "cox")
```
AA Only
```{r}
library(caTools)
library(dplyr)
library(blockForest)
library(survival)
library(randomForestSRC)
library(pec)
#brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
#patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode))
rownames(brca_exp_data) <- patient_ids
normal_aa_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry == "AA"))
normal_aa_samples <- subset(normal_aa_samples, select = -c(genetic_ancestry))
set.seed(757)
split <- sample.split(normal_aa_samples$patient.vital_status, SplitRatio = 0.80)
x_train_rf_aa <- subset(normal_aa_samples, split == TRUE)
x_test_rf_aa <- subset(normal_aa_samples, split == FALSE)
x_train_rf_aa <- subset(x_train_rf_aa, select = -c(tissue_type))
x_test_rf_aa <- subset(x_test_rf_aa, select = -c(tissue_type))
x_train_rf_aa <- subset(x_train_rf_aa, times >= 0)
x_test_rf_aa <- subset(x_test_rf_aa, times >= 0)
rf_aa <- rfsrc(formula =
Surv(times,patient.vital_status)~., data = x_train_rf_aa)
rf_pred_aa <- predict(rf_aa, newdata = x_test_rf_aa)
rf_pred_aa
significant_genes <- c("times", "patient.vital_status", "CD24.100133941", "PRRG1.5638", "IQSEC3.440073","RCC2.55920","CASP8.841")
x_train_cox_aa <- subset(x_train_rf_aa, select = significant_genes)
x_test_cox_aa <- subset(x_test_rf_aa, select = significant_genes)
cox_aa <- coxph(formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
RCC2.55920 + CASP8.841, data = x_train_cox_aa, x = TRUE, y = TRUE)
cox_aa_pec <- pec(cox_aa, formula = Surv(times,patient.vital_status)~ CD24.100133941 + PRRG1.5638 + IQSEC3.440073 + RCC2.55920 + CASP8.841, data = x_test_cox_aa, cens.model = "cox")
crps(cox_aa_pec)
cindex(cox_aa, formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 + RCC2.55920 + CASP8.841, data = x_test_cox_aa, cens.model = "cox")
```
Train EA/AA but test same AA only
```{r}
library(caTools)
library(dplyr)
#library(blockForest)
library(survival)
library(randomForestSRC)
library(pec)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode))
rownames(brca_exp_data) <- patient_ids
normal_ea_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry != "AA"))
normal_ea_samples <- subset(normal_ea_samples, select = -c(genetic_ancestry, tissue_type))
x_train_rf_eaa <- x_train_rf_aa
x_test_rf_eaa <- x_test_rf_aa
x_train_rf_ea <- normal_ea_samples
x_train_rf_eaa <- rbind(x_train_rf_eaa, x_train_rf_ea)
x_train_rf_eaa <- subset(x_train_rf_eaa, times >= 0)
x_test_rf_eaa <- subset(x_test_rf_eaa, times >= 0)
rf_eaa <- rfsrc(formula =
Surv(times,patient.vital_status)~., data = x_train_rf_eaa)
rf_pred_eaa <- predict(rf_eaa, newdata = x_test_rf_eaa)
rf_pred_eaa
significant_genes <- c("times", "patient.vital_status", "CD24.100133941", "PRRG1.5638", "IQSEC3.440073","RCC2.55920","CASP8.841")
x_train_cox_eaa <- subset(x_train_rf_eaa, select = significant_genes)
x_test_cox_eaa <- subset(x_test_rf_eaa, select = significant_genes)
cox_eaa <- coxph(formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
RCC2.55920 + CASP8.841, data = x_train_cox_eaa, x = TRUE, y = TRUE)
cox_eaa_pec <- pec(cox_eaa, formula = Surv(times,patient.vital_status)~ CD24.100133941 + PRRG1.5638 + IQSEC3.440073 + RCC2.55920 + CASP8.841, data = x_test_cox_eaa, cens.model = "cox")
crps(cox_eaa_pec)
cindex(cox_eaa, formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 + RCC2.55920 + CASP8.841, data = x_test_cox_eaa, cens.model = "cox")
plotPredictSurvProb(cox_eaa, newdata = x_test_cox_eaa, xlab = "Survival Time in Days")
```
Train EA/AA, but test EA only
```{r}
normal_aa_samples <- filter(brca_exp_data, (brca_exp_data$genetic_ancestry == "AA"))
x_train_rf_aae <- x_train_rf_ea
x_test_rf_aae <- x_test_rf_ea
x_train_rf_aa <- subset(normal_aa_samples, select = -c(tissue_type, genetic_ancestry))
x_train_rf_aae <- rbind(x_train_rf_aae, x_train_rf_aa)
x_train_rf_aae <- subset(x_train_rf_aae, times >= 0)
x_test_rf_aae <- subset(x_test_rf_aae, times >= 0)
rf_aae <- rfsrc(formula =
Surv(times,patient.vital_status)~., data = x_train_rf_aae)
rf_pred_aae <- predict(rf_aae, newdata = x_test_rf_aae)
rf_pred_aae
significant_genes <- c("times", "patient.vital_status", "CD24.100133941","MRGPRX1.259249", "PRRG1.5638", "IQSEC3.440073","RCC2.55920","CASP8.841")
x_train_cox_aae <- subset(x_train_rf_aae, select = significant_genes)
x_test_cox_aae <- subset(x_test_rf_aae, select = significant_genes)
cox_aae <- coxph(formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 +
RCC2.55920 +
MRGPRX1.259249 + CASP8.841, data = x_train_cox_aae, x = TRUE, y = TRUE)
cox_aae_pec <- pec(cox_aae, formula = Surv(times,patient.vital_status)~ CD24.100133941 + PRRG1.5638 +
MRGPRX1.259249 + IQSEC3.440073 + RCC2.55920 + CASP8.841, data = x_test_cox_aae, cens.model = "cox")
crps(cox_aae_pec)
cindex(cox_aae, formula =
Surv(times,patient.vital_status)~CD24.100133941 + PRRG1.5638 + IQSEC3.440073 + RCC2.55920 + MRGPRX1.259249 + CASP8.841, data = x_test_cox_aae, cens.model = "cox")
```
CNV/mutaations
```{r}
library(TCGA2STAT)
library(dplyr)
library(xml2)
library(parallel)
brca_cnv_all_data <- getTCGA("BRCA",data.type = "CNA_SNP")
brca_cnv_data <- brca_cnv_all_data$dat
class(brca_cnv_data)
brca_cnv_data <- t(brca_cnv_data)
#dim(brca_cnv_data)
#head(brca_cnv_data)
#normal_samples <- filter(brca_cnv_data, #substr(brca_cnv_data$bcr_patient_barcode, 14, 15) == "11")
#dim(normal_samples)
#head(normal_samples)
brca_genetic_ancestry <- read.csv("/Users/andres/Downloads/jp/ancestry.csv", na.strings = "NNN")

ancestries <- c(nrow(brca_cnv_data))

for (i in 1:nrow(brca_cnv_data)) {
  for (j in 1:nrow(brca_genetic_ancestry)) {
    if (grepl(brca_genetic_ancestry$bcr_patient_barcode[j],rownames(brca_cnv_data)[i], fixed = TRUE)){
      ancestries[i] = brca_genetic_ancestry$genetic_ancestry[j]
      break
      }
  }
}
brca_full_data_cnv <- cbind(brca_cnv_data, ancestries)
brca_full_data_cnv <- brca_full_data_cnv[,c(22619,1:(ncol(brca_full_data_cnv) - 1))]

write.csv(brca_full_data_cnv, "brca_cnv_data.csv", row.names=FALSE)

library(RTCGA.clinical)
survival_data <- survivalTCGA(BRCA.clinical)

times_vector <- c(nrow(brca_full_data_cnv))
vital_status_vector <- c(nrow(brca_full_data_cnv))
for (i in 1:nrow(brca_full_data_cnv)) {
  for (j in 1:nrow(survival_data)) {
    if (grepl(survival_data$bcr_patient_barcode[j],rownames(brca_full_data_cnv)[i], fixed = TRUE)){
      times_vector[i] = survival_data$times[j]
      vital_status_vector[i] = survival_data$patient.vital_status[j]
      }
  }
}
brca_full_data_cnv <- cbind(brca_full_data_cnv, times_vector)
brca_full_data_cnv <- brca_full_data_cnv[,c(22620, 1:(ncol(brca_full_data_cnv) - 1))]
brca_full_data_cnv <- cbind(brca_full_data_cnv, vital_status_vector)
brca_full_data_cnv <- brca_full_data_cnv[,c(22621, 1:(ncol(brca_full_data_cnv) - 1))]
head(brca_full_data_cnv)

write.csv(brca_full_data_cnv, "brca_full_data_cna.csv", row.names=TRUE)

tissue_type <- substr(rownames(brca_full_data_cnv), 14, 15) == "01"
tissue_type <- replace(tissue_type,tissue_type==TRUE,"Primary Solid Tumor")
tissue_type <- replace(tissue_type,tissue_type==FALSE,"Solid Tissue Normal")

brca_full_data_cnv_type <- cbind(brca_full_data_cnv, tissue_type)
brca_full_data_cnv_type <- brca_full_data_cnv_type [,c(22622, 1:(ncol(brca_full_data_cnv) - 1))]

write.csv(brca_full_data_cnv_type, "brca_full_data_cna_with_type.csv", row.names=TRUE)
```
CNV Survival
All
```{r}
library(caTools)
library(dplyr)
library(survival)
library(randomForestSRC)
library(pec)
brca_full_cnv_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_cna_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_cnv_data$X
rownames(brca_full_cnv_data) <- patient_ids
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, tissue_type))
brca_cnv_data <- subset(brca_cnv_data, ancestries %in% c("EA", "AA"))
brca_cnv_data <- subset(brca_cnv_data, select = -c(ancestries))
set.seed(757)
split <- sample.split(brca_cnv_data$vital_status_vector, SplitRatio = 0.80)
x_train_rf_cnv <- subset(brca_cnv_data, split == TRUE)
x_test_rf_cnv <- subset(brca_cnv_data, split == FALSE)
x_train_rf_cnv <- subset(x_train_rf_cnv, times_vector >= 0)
x_test_rf_cnv <- subset(x_test_rf_cnv, times_vector >= 0)
rf_cnv_all <- rfsrc(formula =
Surv(times_vector,vital_status_vector)~., data = x_train_rf_cnv)
rf_cnv_pred_all <- predict(rf_cnv_all, newdata = x_test_rf_cnv)
rf_cnv_pred_all
significant_genes <- c("times_vector", "vital_status_vector", "IQSEC3","MRGPRX1","RCC2","CASP8")
x_train_cox_cnv <- subset(x_train_rf_cnv, select = significant_genes)
x_test_cox_cnv <- subset(x_test_rf_cnv, select = significant_genes)
cox_cnv_all <- coxph(formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_train_cox_cnv, x = TRUE, y = TRUE)
cox_cnv_all_pec <- pec(cox_cnv_all, formula = Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv)
crps(cox_cnv_all_pec)
cindex(cox_cnv_all, formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv, cens.model = "cox")
```
CNA EA/AA
```{r}
library(caTools)
library(dplyr)
library(blockForest)
library(survival)
library(randomForestSRC)
library(pec)
#brca_full_cnv_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_cna_with_type.csv", na.strings = "NNN")
#patient_ids <- brca_full_cnv_data$X
#rownames(brca_full_cnv_data) <- patient_ids
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, tissue_type))
normal_ea_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "EA"))
normal_aa_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "AA"))
x_train_rf_cnv_ea_aa <- subset(normal_ea_samples, select = -c(ancestries))
x_test_rf_cnv_ea_aa <- subset(normal_aa_samples, select = -c(ancestries))
x_train_rf_cnv_ea_aa<- subset(x_train_rf_cnv_ea_aa, times_vector >= 0)
x_test_rf_cnv_ea_aa<- subset(x_test_rf_cnv_ea_aa, times_vector >= 0)
rf_cnv_ea_aa <- rfsrc(formula =
Surv(times_vector,vital_status_vector)~., data = x_train_rf_cnv_ea_aa)
rf_cnv_pred_ea_aa <- predict(rf_cnv_ea_aa, newdata = x_test_rf_cnv_ea_aa)
rf_cnv_pred_ea_aa
significant_genes <- c("times_vector", "vital_status_vector", "IQSEC3","MRGPRX1","RCC2","CASP8")
x_train_cox_cnv_ea_aa <- subset(x_train_rf_cnv_ea_aa, select = significant_genes)
x_test_cox_cnv_ea_aa<- subset(x_test_rf_cnv_ea_aa, select = significant_genes)
cox_cnv_ea_aa <- coxph(formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_train_cox_cnv_ea_aa, x = TRUE, y = TRUE)
cox_cnv_ea_aa_pec <- pec(cox_cnv_ea_aa, formula = Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv_ea_aa)
crps(cox_cnv_ea_aa_pec)
cindex(cox_cnv_ea_aa, formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv_ea_aa, cens.model = "cox")
```
EA
```{r}
library(caTools)
library(dplyr)
library(blockForest)
library(survival)
library(randomForestSRC)
library(pec)
#brca_full_cnv_data <- #read.csv("/Users/andres/Downloads/jp/brca_full_data_cna_with_type.csv", na.strings = "NNN")
#patient_ids <- brca_full_cnv_data$X
#rownames(brca_full_cnv_data) <- patient_ids
brca_cnv_data_ea <- filter(brca_full_cnv_data, (brca_full_cnv_data$ancestries == "EA"))
brca_cnv_data_ea <- subset(brca_cnv_data_ea, select = -c(X, ancestries, tissue_type))
set.seed(757)
split <- sample.split(brca_cnv_data_ea$vital_status_vector, SplitRatio = 0.80)
x_train_rf_cnv_ea <- subset(brca_cnv_data_ea, split == TRUE)
x_test_rf_cnv_ea <- subset(brca_cnv_data_ea, split == FALSE)
x_train_rf_cnv_ea <- subset(x_train_rf_cnv_ea, times_vector >= 0)
x_test_rf_cnv_ea <- subset(x_test_rf_cnv_ea, times_vector >= 0)
rf_cnv_ea <- rfsrc(formula =
Surv(times_vector,vital_status_vector)~., data = x_train_rf_cnv_ea)
rf_cnv_ea_pred_all <- predict(rf_cnv_ea, newdata = x_test_rf_cnv_ea)
rf_cnv_ea_pred_all
significant_genes <- c("times_vector", "vital_status_vector", "PRRG1", "IQSEC3","MRGPRX1","RCC2","CASP8")
x_train_cox_cnv_ea <- subset(x_train_rf_cnv_ea, select = significant_genes)
x_test_cox_cnv_ea <- subset(x_test_rf_cnv_ea, select = significant_genes)
cox_cnv_ea <- coxph(formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_train_cox_cnv_ea, x = TRUE, y = TRUE)
cox_cnv_ea_pec <- pec(cox_cnv_ea, formula = Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv_ea)
crps(cox_cnv_ea_pec)
cindex(cox_cnv_ea, formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv_ea, cens.model = "cox")
```
AA
```{r}
library(caTools)
library(dplyr)
library(blockForest)
library(survival)
library(randomForestSRC)
library(pec)
#brca_full_cnv_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_cna_with_type.csv", na.strings = "NNN")
#patient_ids <- brca_full_cnv_data$X
#rownames(brca_full_cnv_data) <- patient_ids
brca_cnv_data_aa <- filter(brca_full_cnv_data, (brca_full_cnv_data$ancestries == "AA"))
brca_cnv_data_aa <- subset(brca_cnv_data_aa, select = -c(X, ancestries, tissue_type))
set.seed(757)
split <- sample.split(brca_cnv_data_aa$vital_status_vector, SplitRatio = 0.80)
x_train_rf_cnv_aa <- subset(brca_cnv_data_aa, split == TRUE)
x_test_rf_cnv_aa <- subset(brca_cnv_data_aa, split == FALSE)
x_train_rf_cnv_aa <- subset(x_train_rf_cnv_aa, times_vector >= 0)
x_test_rf_cnv_aa <- subset(x_test_rf_cnv_aa, times_vector >= 0)
rf_cnv_aa <- rfsrc(formula =
Surv(times_vector,vital_status_vector)~., data = x_train_rf_cnv_aa)
rf_cnv_aa_pred_all <- predict(rf_cnv_aa, newdata = x_test_rf_cnv_aa)
rf_cnv_aa_pred_all
significant_genes <- c("times_vector", "vital_status_vector", "PRRG1", "IQSEC3","MRGPRX1","RCC2","CASP8")
x_train_cox_cnv_aa <- subset(x_train_rf_cnv_aa, select = significant_genes)
x_test_cox_cnv_aa <- subset(x_test_rf_cnv_aa, select = significant_genes)
cox_cnv_aa <- coxph(formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_train_cox_cnv_aa, x = TRUE, y = TRUE)
cox_cnv_aa_pec <- pec(cox_cnv_aa, formula = Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv_aa)
crps(cox_cnv_aa_pec)
cindex(cox_cnv_aa, formula =
Surv(times_vector,vital_status_vector)~IQSEC3 +
MRGPRX1 + RCC2 + CASP8, data = x_test_cox_cnv_aa, cens.model = "cox")
```
Train AA/EA, Test AA
```{r}
normal_ea_samples <- filter(brca_full_cnv_data, (brca_full_cnv_data$ancestries == "EA"))
normal_ea_samples <- subset(normal_ea_samples, select = -c(X, ancestries, tissue_type))
x_train_rf_cnv_eaa <- x_train_rf_cnv_aa
x_test_rf_cnv_eaa <- x_test_rf_cnv_aa
x_train_rf_cnv_ea <- normal_ea_samples
x_train_rf_cnv_eaa <- rbind(x_train_rf_cnv_eaa, x_train_rf_cnv_ea)
x_train_rf_cnv_eaa <- subset(x_train_rf_cnv_eaa, times_vector >= 0)
x_test_rf_cnv_eaa <- subset(x_test_rf_cnv_eaa, times_vector >= 0)
rf_cnv_eaa <- rfsrc(formula =
Surv(times_vector,vital_status_vector)~., data = x_train_rf_cnv_eaa)
rf_cnv_pred_eaa <- predict(rf_cnv_eaa, newdata = x_test_rf_cnv_eaa)
rf_cnv_pred_eaa
significant_genes <- c("times_vector", "vital_status_vector", "IQSEC3","RCC2","CASP8", "MRGPRX1")
x_train_cox_cnv_eaa <- subset(x_train_rf_cnv_eaa, select = significant_genes)
x_test_cox_cnv_eaa <- subset(x_test_rf_cnv_eaa, select = significant_genes)
cox_cnv_eaa <- coxph(formula =
Surv(times_vector, vital_status_vector)~IQSEC3 +
RCC2 + CASP8 + MRGPRX1, data = x_train_cox_cnv_eaa, x = TRUE, y = TRUE)
cox_cnv_eaa_pec <- pec(cox_cnv_eaa, formula = Surv(times_vector, vital_status_vector)~MRGPRX1 + IQSEC3 + RCC2 + CASP8, data = x_test_cox_cnv_eaa, cens.model = "cox")
crps(cox_cnv_eaa_pec)
cindex(cox_cnv_eaa, formula =
Surv(times_vector, vital_status_vector)~MRGPRX1 + IQSEC3 + RCC2 + CASP8, data = x_test_cox_cnv_eaa, cens.model = "cox")

```
Train EA/AA, test EA
```{r}
normal_aa_samples <- filter(brca_full_cnv_data, (brca_full_cnv_data$ancestries == "AA"))
normal_aa_samples <- subset(normal_aa_samples, select = -c(X, ancestries, tissue_type))
x_train_rf_cnv_aae <- x_train_rf_cnv_ea
x_test_rf_cnv_aae <- x_test_rf_cnv_ea
x_train_rf_cnv_aa <- normal_aa_samples
x_train_rf_cnv_aae <- rbind(x_train_rf_cnv_aae, x_train_rf_cnv_aa)
x_train_rf_cnv_aae <- subset(x_train_rf_cnv_aae, times_vector >= 0)
x_test_rf_cnv_aae <- subset(x_test_rf_cnv_aae, times_vector >= 0)
rf_cnv_aae <- rfsrc(formula =
Surv(times_vector,vital_status_vector)~., data = x_train_rf_cnv_aae)
rf_cnv_pred_aae <- predict(rf_cnv_aae, newdata = x_test_rf_cnv_aae)
rf_cnv_pred_aae
significant_genes <- c("times_vector", "vital_status_vector", "IQSEC3","RCC2","CASP8", "MRGPRX1")
x_train_cox_cnv_aae <- subset(x_train_rf_cnv_aae, select = significant_genes)
x_test_cox_cnv_aae <- subset(x_test_rf_cnv_aae, select = significant_genes)
cox_cnv_aae <- coxph(formula =
Surv(times_vector, vital_status_vector)~IQSEC3 +
RCC2 + CASP8 + MRGPRX1, data = x_train_cox_cnv_aae, x = TRUE, y = TRUE)
cox_cnv_aae_pec <- pec(cox_cnv_aae, formula = Surv(times_vector, vital_status_vector)~MRGPRX1 + IQSEC3 + RCC2 + CASP8, data = x_test_cox_cnv_aae, cens.model = "cox")
crps(cox_cnv_aae_pec)
cindex(cox_cnv_aae, formula =
Surv(times_vector, vital_status_vector)~MRGPRX1 + IQSEC3 + RCC2 + CASP8, data = x_test_cox_cnv_aae, cens.model = "cox")

```
PCA Analysis
```{r}
library(dplyr)
library(DESeq2)
brca_full_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_data$bcr_patient_barcode
rownames(brca_full_data) <- patient_ids
brca_exp_data <- subset(brca_full_data, select = -c(bcr_patient_barcode, times, patient.vital_status))
brca_exp_data <- subset(brca_exp_data, genetic_ancestry %in% c("EA", "AA"))
brca_exp_data <- subset(brca_exp_data, select = -c(genetic_ancestry))
brca_clin_data <- subset(brca_full_data, select = c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status, tissue_type))
brca_clin_data <- subset(brca_clin_data, genetic_ancestry %in% c("EA", "AA"))
brca_exp_data_t <- t(brca_exp_data)
misclassified_indices <- which(brca_clin_data$bcr_patient_barcode %in% c("TCGA-BH-A1FE-06A-11R-A213-07", "TCGA-E2-A14U-01A-11R-A22K-07", "TCGA-E9-A1N5-11A-41R-A14D-07", "TCGA-A2-A0CQ-01A-21R-A034-07", "TCGA-E2-A1LL-01A-11R-A144-07", "TCGA-PL-A8LY-01A-11R-A41B-07", "TCGA-AR-A0TY-01A-12R-A115-07", "TCGA-AR-A0TY-01A-12R-A115-07", "TCGA-E9-A1N5-11A-41R-A14D-07", "TCGA-BH-A0DK-11A-13R-A089-07",
"TCGA-BH-A0DZ-11A-22R-A089-07",
"TCGA-E9-A1N5-11A-41R-A14D-07", "TCGA-BH-A0BW-11A-12R-A115-07", "TCGA-BH-A42U-01A-12R-A24H-07","TCGA-PL-A8LY-01A-11R-A41B-07",
"TCGA-A2-A1G6-01A-11R-A13Q-07", 
"TCGA-AR-A0TS-01A-11R-A115-07", "TCGA-BH-A0DZ-11A-22R-A089-07"))
classified_vector <- rep("Correctly Classified",nrow(brca_clin_data))
classified_vector[misclassified_indices] <- "Incorrectly Classified"
brca_metadata <- cbind(brca_clin_data$bcr_patient_barcode, brca_clin_data$genetic_ancestry, brca_clin_data$tissue_type, classified_vector)
colnames(brca_metadata) <- c("bcr_patient_barcode","genetic_ancestry","tissue_type","classified_vector")
gene_names_exp <- rownames(brca_exp_data_t)
brca_exp_data_t <- as.data.frame(brca_exp_data_t)
#row.names(brca_exp_data_t) <- gene_names_exp
#colnames(brca_exp_data_t) <- NULL
brca_exp_data_t <- brca_exp_data_t[1:20531,]
copy_t <- brca_exp_data_t
brca_exp_data_t <- as.data.frame(sapply(brca_exp_data_t, as.numeric))
filtered_exp_data <- filter(brca_exp_data_t, (rowSums(brca_exp_data_t) >= 10))
filtered_expression_df <- round(filtered_exp_data)

dds <- DESeqDataSetFromMatrix(
  countData = filtered_expression_df, 
  colData = brca_metadata,
  design = ~tissue_type
)

dds_norm <- vst(dds)

plotPCA(
  dds_norm,
  intgroup = c("tissue_type","classified_vector")
)
```
PCA of AA/EA
```{r}
plotPCA(
  dds_norm,
  intgroup = c("tissue_type","genetic_ancestry")
)
```

kaplain meier curve
```{r}
library(survival)
library(ggplot2)
library(ggfortify)
brca_clin_data <- subset(brca_full_data, select = c(bcr_patient_barcode, genetic_ancestry, times, patient.vital_status))
brca_clin_data <- subset(brca_clin_data, genetic_ancestry %in% c("EA", "AA"))
surv_obj <- Surv(brca_clin_data$times, brca_clin_data$patient.vital_status)
model_fit <- survfit(surv_obj ~ 1, data = brca_clin_data)
autoplot(model_fit) + 
 labs(x = "\n Survival Time (Days) ", y = "Survival Probabilities \n", 
 title = "Survival Times Of TCGA BRCA Patients \n Stratified by Race") + 
 theme(plot.title = element_text(hjust = 0.5), 
 axis.title.x = element_text(face="bold", colour="#FF7A33", size = 12),
 axis.title.y = element_text(face="bold", colour="#FF7A33", size = 12),
 legend.title = element_text(face="bold", size = 10))
```

```{r}
library(ggsurvfit)
survfit2(surv_obj ~ genetic_ancestry, data = brca_clin_data)%>% 
  ggsurvfit() +
  labs(
    x = "Days",
    y = "Overall survival probability"
  ) + 
  add_confidence_interval() +
  add_risktable()

aa_times <- brca_clin_data$patient.vital_status[which(brca_clin_data$genetic_ancestry == "AA")]
length(which(aa_times == "1"))
```
Plot RF Predicted Curves
```{r}
# Plot RF Predicted Curves
library(randomForestSRC)
library(pec)
library(survival)
library(ggplot2)
rf_cnv_aae <- readRDS("/Users/andres/Downloads/jp/rf_cnv_aae.RDS")
x_test_rf_cnv_aae <- readRDS("/Users/andres/Downloads/jp/x_test_rf_cnv_aae.RDS")
x_train_rf_cnv_aae <- readRDS("/Users/andres/Downloads/jp/x_train_rf_cnv_aae.RDS")
plotPredictSurvProb(rf_cnv_aae, newdata = x_test_rf_cnv_aae, xlab = "Survival Time in Days")
```
CNV Classification
```{r}
library(caret)
library(caTools)
library(dplyr)

brca_full_cnv_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_cna_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_cnv_data$X
rownames(brca_full_cnv_data) <- patient_ids
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector))
brca_cnv_data <- subset(brca_cnv_data, ancestries %in% c("EA", "AA"))
brca_cnv_data <- subset(brca_cnv_data, select = -c(ancestries))
brca_cnv_data$tissue_type <- as.factor(brca_cnv_data$tissue_type)
set.seed(757)
split <- sample.split(brca_cnv_data$tissue_type, SplitRatio = 0.80)

all_train_cnv_svm <- subset(brca_cnv_data, split == TRUE)
all_test_cnv_svm <- subset(brca_cnv_data, split == FALSE)


fit_svm_all_cnv <- train(tissue_type ~ ., data = all_train_cnv_svm, method = "svmLinear")
predict_svm_all_cnv <- predict(fit_svm_all_cnv, all_test_cnv_svm)
confusionMatrix(reference=all_test_cnv_svm$tissue_type, data = predict_svm_all_cnv, mode = "everything")
predictions_all_cnv <- as.character(predict_svm_all_cnv)

rownames(all_test_cnv_svm)[which(all_test_cnv_svm$tissue_type != predictions_all_cnv)]
all_aa_test_cnv <- subset(brca_full_cnv_data, X %in% rownames(all_test_cnv_svm))
length(which(all_aa_test_cnv$ancestries == "AA"))

```
EA/AA
```{r}
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector))
brca_cnv_data$tissue_type <- as.factor(brca_cnv_data$tissue_type)

normal_ea_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "EA"))
normal_aa_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "AA"))

ea_aa_train_cnv_svm <- subset(normal_ea_samples, select = -c(ancestries))
ea_aa_test_cnv_svm <- subset(normal_aa_samples, select = -c(ancestries))

fit_svm_ea_aa_cnv <- train(tissue_type ~ ., data = ea_aa_train_cnv_svm, method = "svmLinear")
predict_svm_ea_aa_cnv <- predict(fit_svm_ea_aa_cnv, ea_aa_test_cnv_svm)
confusionMatrix(reference=ea_aa_test_cnv_svm$tissue_type, data = predict_svm_ea_aa_cnv, mode = "everything")
predictions_ea_aa_cnv <- as.character(predict_svm_ea_aa_cnv)

rownames(ea_aa_test_cnv_svm)[which(ea_aa_test_cnv_svm$tissue_type != predictions_ea_aa_cnv)]

```
EA Only
```{r}
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector))
brca_cnv_data$tissue_type <- as.factor(brca_cnv_data$tissue_type)

normal_ea_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "EA"))

normal_ea_samples <- subset(normal_ea_samples, select = -c(ancestries))

split <- sample.split(normal_ea_samples$tissue_type, SplitRatio = 0.80)

ea_train_cnv_svm <- subset(normal_ea_samples, split == TRUE)
ea_test_cnv_svm <- subset(normal_ea_samples, split == FALSE)


fit_svm_ea_cnv <- train(tissue_type ~ ., data = ea_train_cnv_svm, method = "svmLinear")
predict_svm_ea_cnv <- predict(fit_svm_ea_cnv, ea_test_cnv_svm)
confusionMatrix(reference=ea_test_cnv_svm$tissue_type, data = predict_svm_ea_cnv, mode = "everything")
predictions_ea_cnv <- as.character(predict_svm_ea_cnv)

rownames(ea_test_cnv_svm)[which(ea_test_cnv_svm$tissue_type != predictions_ea_cnv)]
```
AA Only
```{r}
normal_aa_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "AA"))

normal_aa_samples <- subset(normal_aa_samples, select = -c(ancestries))

split <- sample.split(normal_aa_samples$tissue_type, SplitRatio = 0.80)

aa_train_cnv_svm <- subset(normal_aa_samples, split == TRUE)
aa_test_cnv_svm <- subset(normal_aa_samples, split == FALSE)


fit_svm_aa_cnv <- train(tissue_type ~ ., data = aa_train_cnv_svm, method = "svmLinear")
predict_svm_aa_cnv <- predict(fit_svm_aa_cnv, aa_test_cnv_svm)
confusionMatrix(reference=aa_test_cnv_svm$tissue_type, data = predict_svm_aa_cnv, mode = "everything")
predictions_aa_cnv <- as.character(predict_svm_aa_cnv)

rownames(aa_test_cnv_svm)[which(aa_test_cnv_svm$tissue_type != predictions_aa_cnv)]

```
Trained EA/AA, tested same AA
```{r}
eaa_train_cnv_svm <- rbind(normal_ea_samples, aa_train_cnv_svm)
fit_svm_eaa_cnv <- train(tissue_type ~ ., data = eaa_train_cnv_svm, method = "svmLinear")
predict_svm_eaa_cnv <- predict(fit_svm_eaa_cnv, aa_test_cnv_svm)
confusionMatrix(reference=aa_test_cnv_svm$tissue_type, data = predict_svm_eaa_cnv, mode = "everything")

predictions_eaa <- as.character(predict_svm_eaa_cnv)
rownames(aa_test_cnv_svm)[which(aa_test_cnv_svm$tissue_type != predictions_eaa)]

#fit_svm_eaa_cnv <- readRDS("fit_svm_eaa_cnv.RDS")
```
Trained EA/AA, tested same EA
```{r}
aae_train_cnv_svm <- rbind(ea_train_cnv_svm, aa_train_cnv_svm, aa_test_cnv_svm)
fit_svm_aae_cnv <- train(tissue_type ~ ., data = aae_train_cnv_svm, method = "svmLinear")
predict_svm_aae_cnv <- predict(fit_svm_aae_cnv, ea_test_cnv_svm)
confusionMatrix(reference=ea_test_cnv_svm$tissue_type, data = predict_svm_aae_cnv, mode = "everything")

predictions_aae <- as.character(predict_svm_aae_cnv)
rownames(ea_test_cnv_svm)[which(ea_test_cnv_svm$tissue_type != predictions_aae)]

```
Glmnet CNV
```{r}
brca_full_cnv_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_cna_with_type.csv", na.strings = "NNN")
patient_ids <- brca_full_cnv_data$X
rownames(brca_full_cnv_data) <- patient_ids
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector))
brca_cnv_data$tissue_type <- as.factor(brca_cnv_data$tissue_type)

set.seed(13)
brca_cnv_data <- subset(brca_cnv_data, select = -c(ancestries))
split <- sample.split(brca_cnv_data$tissue_type, SplitRatio = 0.80)
all_train_glm_cnv <- subset(brca_cnv_data, split == TRUE)
all_test_glm_cnv <- subset(brca_cnv_data, split == FALSE)
y_train_glm_cnv <- as.matrix(all_train_glm_cnv$tissue_type)
x_train_glm_cnv <- as.matrix(subset(all_train_glm_cnv, select = -c(tissue_type)))
res_cnv = cv.glmnet(
  x = x_train_glm_cnv,
  y = y_train_glm_cnv,
  alpha = 0.5,
  family = "binomial"
)
y_test_glm_cnv <- as.matrix(all_test_glm_cnv$tissue_type)
x_test_glm_cnv <- as.matrix(subset(all_test_glm_cnv, select = -c(tissue_type)))

y_pred_glm_cnv = predict(res_cnv, newx=x_test_glm_cnv, type="class", s="lambda.min")
confusion_matrix = table(y_pred_glm_cnv, y_test_glm_cnv)
print(confusion_matrix)
rownames(y_pred_glm_cnv)[which(y_test_glm_cnv != y_pred_glm_cnv)]

all_aa_test_g <- subset(brca_full_cnv_data, X %in% rownames(all_test_glm_cnv))
length(which(all_aa_test_g$ancestries == "AA"))

```
EA/AA
```{r}
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector))
brca_cnv_data$tissue_type <- as.factor(brca_cnv_data$tissue_type)

normal_aa_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "AA"))
normal_ea_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "EA"))


all_train_glm_cnv <- subset(normal_ea_samples, select = -c(ancestries))
all_test_glm_cnv <- subset(normal_aa_samples, select = -c(ancestries))

y_train_glm_cnv <- as.matrix(all_train_glm_cnv$tissue_type)
x_train_glm_cnv <- as.matrix(subset(all_train_glm_cnv, select = -c(tissue_type)))
res_cnv = cv.glmnet(
  x = x_train_glm_cnv,
  y = y_train_glm_cnv,
  alpha = 0.5,
  family = "binomial"
)

y_test_glm_cnv <- as.matrix(all_test_glm_cnv$tissue_type)
x_test_glm_cnv <- as.matrix(subset(all_test_glm_cnv, select = -c(tissue_type)))

y_pred_glm_cnv = predict(res_cnv, newx=x_test_glm_cnv, type="class", s="lambda.min")
confusion_matrix = table(y_pred_glm_cnv, y_test_glm_cnv)
print(confusion_matrix)
rownames(y_pred_glm_cnv)[which(y_test_glm_cnv != y_pred_glm_cnv)]

all_aa_test_g <- subset(brca_full_cnv_data, X %in% rownames(all_test_glm_cnv))
length(which(all_aa_test_g$ancestries == "AA"))


```
EA Only
```{r}
brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector))
brca_cnv_data$tissue_type <- as.factor(brca_cnv_data$tissue_type)

normal_ea_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "EA"))

split <- sample.split(normal_ea_samples$tissue_type, SplitRatio = 0.80)

all_train_glm_cnv_ea <- subset(normal_ea_samples, split == TRUE)
all_test_glm_cnv_ea <- subset(normal_ea_samples, split == FALSE)

all_train_glm_cnv_ea <- subset(all_train_glm_cnv_ea, select = -c(ancestries))
all_test_glm_cnv_ea <- subset(all_test_glm_cnv_ea, select = -c(ancestries))


y_train_glm_cnv_ea <- as.matrix(all_train_glm_cnv_ea$tissue_type)
x_train_glm_cnv_ea <- as.matrix(subset(all_train_glm_cnv_ea, select = -c(tissue_type)))

res_cnv_ea = cv.glmnet( x = x_train_glm_cnv_ea, y = y_train_glm_cnv_ea, alpha = 0.5, family = "binomial")

y_test_glm_cnv_ea <- as.matrix(all_test_glm_cnv_ea$tissue_type)
x_test_glm_cnv_ea <- as.matrix(subset(all_test_glm_cnv_ea, select = -c(tissue_type)))

y_pred_glm_cnv_ea = predict(res_cnv_ea, newx=x_test_glm_cnv_ea, type="class", s="lambda.min")
confusion_matrix = table(y_pred_glm_cnv_ea, y_test_glm_cnv_ea)
print(confusion_matrix)
rownames(y_pred_glm_cnv_ea)[which(y_test_glm_cnv_ea != y_pred_glm_cnv_ea)]

```

AA
```{r}
normal_aa_samples <- filter(brca_cnv_data, (brca_cnv_data$ancestries == "AA"))

split <- sample.split(normal_aa_samples$tissue_type, SplitRatio = 0.80)

all_train_glm_cnv_aa <- subset(normal_aa_samples, split == TRUE)
all_test_glm_cnv_aa <- subset(normal_aa_samples, split == FALSE)

all_train_glm_cnv_aa <- subset(all_train_glm_cnv_aa, select = -c(ancestries))
all_test_glm_cnv_aa <- subset(all_test_glm_cnv_aa, select = -c(ancestries))


y_train_glm_cnv_aa <- as.matrix(all_train_glm_cnv_aa$tissue_type)
x_train_glm_cnv_aa <- as.matrix(subset(all_train_glm_cnv_aa, select = -c(tissue_type)))

res_cnv_aa = cv.glmnet(x = x_train_glm_cnv_aa, y = y_train_glm_cnv_aa, alpha = 0.5, family = "binomial")

y_test_glm_cnv_aa <- as.matrix(all_test_glm_cnv_aa$tissue_type)
x_test_glm_cnv_aa <- as.matrix(subset(all_test_glm_cnv_aa, select = -c(tissue_type)))

y_pred_glm_cnv_aa = predict(res_cnv_aa, newx=x_test_glm_cnv_aa, type="class", s="lambda.min")
confusion_matrix = table(y_pred_glm_cnv_aa, y_test_glm_cnv_aa)
print(confusion_matrix)
rownames(y_pred_glm_cnv_aa)[which(y_test_glm_cnv_aa != y_pred_glm_cnv_aa)]
```
EAA
```{r}
eaa_x_train_cnv <- rbind(x_train_glm_cnv_ea, x_train_glm_cnv_aa, x_test_glm_cnv_ea)
eaa_y_train_cnv <- rbind(y_train_glm_cnv_ea, y_train_glm_cnv_aa, y_test_glm_cnv_ea)
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)

res_cnv_eaa = cv.glmnet(x = eaa_x_train_cnv, y = eaa_y_train_cnv, alpha = 0.5, family = "binomial")

y_test_glm_cnv_eaa <- y_test_glm_cnv_aa
x_test_glm_cnv_eaa <- x_test_glm_cnv_aa

y_pred_glm_cnv_eaa = predict(res_cnv_eaa, newx=x_test_glm_cnv_eaa, type="class", s="lambda.min")
confusion_matrix = table(y_pred_glm_cnv_eaa, y_test_glm_cnv_eaa)
print(confusion_matrix)
rownames(y_pred_glm_cnv_eaa)[which(y_test_glm_cnv_eaa != y_pred_glm_cnv_eaa)]

```
AAE
```{r}
aae_x_train_cnv <- rbind(x_train_glm_cnv_ea, x_train_glm_cnv_aa, x_test_glm_cnv_aa)
aae_y_train_cnv <- rbind(y_train_glm_cnv_ea, y_train_glm_cnv_aa, y_test_glm_cnv_aa)
#brca_exp_data$tissue_type <- as.factor(brca_exp_data$tissue_type)

res_cnv_aae = cv.glmnet(x = aae_x_train_cnv, y = aae_y_train_cnv, alpha = 0.5, family = "binomial")

y_test_glm_cnv_aae <- y_test_glm_cnv_ea
x_test_glm_cnv_aae <- x_test_glm_cnv_ea

y_pred_glm_cnv_aae = predict(res_cnv_aae, newx=x_test_glm_cnv_aae, type="class", s="lambda.min")
confusion_matrix = table(y_pred_glm_cnv_aae, y_test_glm_cnv_aae)
print(confusion_matrix)
rownames(y_pred_glm_cnv_aae)[which(y_test_glm_cnv_aae != y_pred_glm_cnv_aae)]
```

CNV PCA
```{r}
library(dplyr)
library(DESeq2)
library(M3C)
library(ggplot2)
library(ggfortify)

brca_full_cnv_data <- read.csv("/Users/andres/Downloads/jp/brca_full_data_cna_with_type.csv", na.strings = "NNN")

brca_clin_data_cnv <- subset(brca_full_cnv_data, select = c(X, ancestries, tissue_type, times_vector, vital_status_vector))
brca_clin_data_cnv <- subset(brca_clin_data_cnv, ancestries %in% c("EA", "AA"))

brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector, tissue_type))
brca_cnv_data <- subset(brca_cnv_data, ancestries %in% c("EA", "AA"))
brca_cnv_data <- subset(brca_cnv_data, select = -c(ancestries))


brca_cnv_data_t <- t(brca_cnv_data)
misclassified_indices_cnv <- which(brca_cnv_data_f$X %in% c("TCGA-A1-A0SB-01A-11D-A141-01","TCGA-A2-A0ST-01A-12D-A087-01","TCGA-AN-A0XV-01A-11D-A107-01","TCGA-BH-A0BR-01A-21D-A111-01", "TCGA-BH-A1FU-01A-11D-A14F-01", "TCGA-E2-A1B6-01A-31D-A12N-01","TCGA-EW-A1P1-01A-31D-A14F-01","TCGA-GM-A2DB-01A-31D-A18N-01","TCGA-GM-A2DI-01A-31D-A18N-01","TCGA-GM-A3XG-01A-31D-A242-01","TCGA-A2-A1G6-01A-11D-A13J-01","TCGA-A7-A3RF-01A-11D-A227-01","TCGA-AN-A0FS-10A-01D-A037-01","TCGA-B6-A0IG-01A-11D-A036-01","TCGA-BH-A1ES-01A-11D-A134-01","TCGA-D8-A1JH-01A-11D-A13J-01","TCGA-D8-A27E-01A-11D-A16C-01","TCGA-E2-A14U-01A-11D-A227-01","TCGA-E2-A1IU-01A-11D-A14F-01","TCGA-A2-A0CL-01A-11D-A111-01","TCGA-A2-A25E-01A-11D-A166-01","TCGA-A2-A3XW-01A-11D-A238-01","TCGA-A7-A2KD-01A-31D-A21P-01","TCGA-A7-A4SF-01A-11D-A25N-01","TCGA-AO-A12H-01A-11D-A111-01","TCGA-B6-A3ZX-01A-11D-A238-01","TCGA-E2-A572-01A-13D-A31T-01","TCGA-LD-A9QF-01A-32D-A41E-01","TCGA-OL-A5RW-10A-01D-A28D-01","TCGA-OL-A66H-01A-11D-A29M-01","TCGA-OL-A66O-01A-11D-A31T-01","TCGA-OL-A66P-01A-11D-A31T-01","TCGA-PL-A8LX-01A-11D-A41E-01","TCGA-PL-A8LY-01A-11D-A41E-01","TCGA-S3-AA15-01A-11D-A41E-01", "TCGA-AC-A8OR-01A-21D-A41E-01","TCGA-AO-A1KO-01A-31D-A13J-01","TCGA-AQ-A0Y5-01A-11D-A14J-01","TCGA-AR-A1AO-01A-11D-A12N-01","TCGA-BH-A0H5-01A-21D-A111-01","TCGA-D8-A1XA-01A-11D-A14F-01","TCGA-E9-A5UO-01A-11D-A28A-01","TCGA-BH-A42V-01A-11D-A242-01","TCGA-GM-A2DO-01A-11D-A18N-01","TCGA-A2-A0ER-01A-21D-A036-01","TCGA-A2-A4S0-01A-21D-A25N-01","TCGA-AO-A0JC-01A-11D-A059-01","TCGA-B6-A0RO-01A-22D-A087-01","TCGA-BH-A0H7-11A-13D-A092-01",  "TCGA-E9-A2JT-01A-22D-A18N-01","TCGA-A2-A0CR-01A-11D-A227-01","TCGA-A2-A0ES-01A-11D-A111-01","TCGA-A2-A0ST-01A-12D-A087-01","TCGA-A8-A082-01A-11D-A011-01","TCGA-A8-A08H-01A-21D-A011-01","TCGA-AC-A2FK-01A-12D-A17U-01", "TCGA-AO-A0JC-01A-11D-A059-01","TCGA-AR-A1AO-01A-11D-A12N-01","TCGA-BH-A1EW-11B-33D-A134-01","TCGA-BH-A1EX-01A-11D-A13J-01","TCGA-BH-A1FU-01A-11D-A14F-01","TCGA-BH-A28O-01A-11D-A227-01","TCGA-E9-A2JT-01A-22D-A18N-01","TCGA-EW-A1P1-01A-31D-A14F-01","TCGA-EW-A1PF-01A-11D-A141-01","TCGA-OL-A66H-01A-11D-A29M-01","TCGA-A2-A0CL-01A-11D-A111-01","TCGA-A2-A3XV-01A-21D-A238-01","TCGA-A2-A3XW-01A-11D-A238-01","TCGA-AO-A12H-01A-11D-A111-01","TCGA-BH-A42U-01A-12D-A242-01","TCGA-E2-A1LB-01A-11D-A141-01","TCGA-E2-A1LL-01A-11D-A141-01","TCGA-LD-A9QF-01A-32D-A41E-01","TCGA-OL-A5RY-01A-21D-A28A-01","TCGA-OL-A66P-01A-11D-A31T-01","TCGA-PL-A8LX-01A-11D-A41E-01","TCGA-PL-A8LY-01A-11D-A41E-01","TCGA-S3-AA15-01A-11D-A41E-01","TCGA-A1-A0SB-01A-11D-A141-01","TCGA-AN-A0XV-01A-11D-A107-01","TCGA-BH-A0BR-01A-21D-A111-01",  "TCGA-E2-A1B6-01A-31D-A12N-01","TCGA-GM-A2DB-01A-31D-A18N-01","TCGA-GM-A2DI-01A-31D-A18N-01","TCGA-GM-A3XG-01A-31D-A242-01"))
classified_vector_cnv <- rep("Correctly Classified",nrow(brca_cnv_data_f))
classified_vector_cnv[misclassified_indices_cnv] <- "Incorrectly Classified"
brca_metadata_cnv <- cbind(brca_cnv_data_f$X, brca_cnv_data_f$ancestries, brca_cnv_data_f$tissue_type, classified_vector_cnv)

colnames(brca_metadata_cnv) <- c("bcr_patient_barcode","genetic_ancestry","tissue_type","classified_vector")

brca_cnv_data_t <- as.data.frame(brca_cnv_data_t)

#brca_exp_data_t <- brca_exp_data_t[1:20531,]

brca_cnv_data_f <- subset(brca_full_cnv_data, ancestries %in% c("EA", "AA"))
brca_cnv_data_f[] <- classified_vector_cnv

cnv_pca <- prcomp(brca_cnv_data)
cnv_pca <- readRDS("cnv_pca.RDS")
brca_cnv_data_f %>% unite('type_ancestry', tissue_type, ancestries, sep=':') -> brca_cnv_data_f
autoplot(cnv_pca, data = brca_cnv_data_f, colour = 'type_ancestry')

brca_cnv_data <- subset(brca_full_cnv_data, select = -c(X, times_vector, vital_status_vector, tissue_type))
brca_cnv_data <- subset(brca_cnv_data, ancestries %in% c("EA", "AA"))


pca(cnv_df,colvec=c('skyblue','tomato'),
labels=brca_metadata_cnv[,"classified_vector"],controlscale=TRUE,scale=3,printres=TRUE,printwidth=25)

dds_cnv <- DESeqDataSetFromMatrix(
  countData = cnv_df, 
  colData = brca_metadata_cnv,
  design = ~tissue_type
)

dds_norm_cnv <- vst(dds_cnv)

plotPCA(
  dds_norm_cnv,
  intgroup = c("tissue_type","classified_vector")
)

```







